### DISCUSSION TITLE
Why is bevy::core::Label a COW String?

### DISCUSSION THREAD

**Discussion by sdfgeoff** [bevyengine/bevy#3929] [Q&A]
I'm working on the [blender_bevy_exporter](https://github.com/sdfgeoff/blender_bevy_toolkit) which generates bevy scene files using Blender3D. I'd like to minimize the number of ""special" component types and use core types as much as possible (for meshes + materials, custom "loader" types are fine). 

As such, I'd like to copy blender's name for objects into the "Name" field, but upon serializing an entity I get:
```
      {
        "type": "bevy_core::name::Name",
        "struct": {
          "hash": {
            "type": "u64",
            "value": 2656167464584161299,
          },
          "name": {
            "type": "alloc::borrow::Cow<str>",
            "value": "Test",
          },
        },
      },

```

I'd just like to raise the question of why the name struct is anything other than a plain-old "String"? It means i've defined my own:
```
/// Component that contains the name of the object
#[derive(Reflect, Default, Component, Debug)]
#[reflect(Component)]
pub struct BlendLabel {
    pub name: String,
}
```
Which feels a bit odd.

**Comment by bjorn3** [comment]
A `Cow<str>` allows avoiding an allocation if the string is fixed at compile time. The hash field exists to allow for faster comparisons I believe.

**Comment by mockersf** [comment]
the `Name` serialisation / deserialisation should be customised to not show the hash, it could expose itself as a simple String.
https://github.com/bevyengine/bevy/blob/d8974e7c3d84189313268f475ceeff9393cedffa/crates/bevy_core/src/name.rs#L16-L21

**Reply by cart** [reply]
Agreed
